### PR TITLE
SDD 0021 - Cluster Catalog Compilation

### PR DESCRIPTION
**Pull request by srueg**
* Rewording of the "Compilation Engine"
* AuthN & AuthZ for the compilation engine

### CODE REVIEW

**Comment by tobru** [comment]
LGTM

File: docs/modules/SDDs/pages/0021-cluster-catalog-compilation.adoc [hunk 1]
```diff
@@ -92,14 +92,14 @@ A permission system needs to be in place to decide which entities are allowed to
...
+=== Compilation Engine
 
 To compile the catalog for a cluster, a compilation job is created.
 The compilation job is a https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion[Kubernetes job] resource which executes the xref:0006-commodore-configuration-catalog-compilation.adoc[catalog compilation engine].
+The compilation engine starts a compilation job if the `cluster.syn.tools/compilation-id` annotation has changed on a cluster object.
 
-To distribute the load this compilation generates and to follow a decentralized approach, these jobs are run on the same cluster as they compile the catalog for.
-The Steward component on a cluster starts a compilation job if the `cluster.syn.tools/needs-compilation` annotation is set to `true` on its own cluster object.
-It will then launch the respective job to compile the cluster catalog.
+To distribute the load the compilation generates and to follow a decentralized approach, these jobs are run on the same cluster as they compile the catalog for.
```
Comment (corvus-ch) [inline docs/modules/SDDs/pages/0021-cluster-catalog-compilation.adoc]: That sentences is missing something. Also the decentralization is somewhat redundant with the next sentence.

**Review by corvus-ch — CHANGES_REQUESTED** [review]

File: docs/modules/SDDs/pages/0021-cluster-catalog-compilation.adoc [hunk 2]
```diff
@@ -92,14 +92,14 @@ A permission system needs to be in place to decide which entities are allowed to
...
+=== Compilation Engine
 
 To compile the catalog for a cluster, a compilation job is created.
 The compilation job is a https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion[Kubernetes job] resource which executes the xref:0006-commodore-configuration-catalog-compilation.adoc[catalog compilation engine].
+The compilation engine starts a compilation job if the `cluster.syn.tools/compilation-id` annotation has changed on a cluster object.
 
-To distribute the load this compilation generates and to follow a decentralized approach, these jobs are run on the same cluster as they compile the catalog for.
-The Steward component on a cluster starts a compilation job if the `cluster.syn.tools/needs-compilation` annotation is set to `true` on its own cluster object.
-It will then launch the respective job to compile the cluster catalog.
+By default the compilation engine creates the compile jobs on the respective target clusters (for example via Steward).
```
Comment (corvus-ch) [inline docs/modules/SDDs/pages/0021-cluster-catalog-compilation.adoc]:
```suggestion
Following a decentralized approach, the compilation engine creates the compile jobs on the respective target clusters (for example via Steward).
```

File: docs/modules/SDDs/pages/0021-cluster-catalog-compilation.adoc [hunk 3]
```diff
@@ -92,14 +92,14 @@ A permission system needs to be in place to decide which entities are allowed to
...
+With the goal to distribute the load the compilation generates and to follow a decentralized approach.
```
Comment (corvus-ch) [inline docs/modules/SDDs/pages/0021-cluster-catalog-compilation.adoc]:
```suggestion
This removes the need for a central infrastructure which would do nothing most of the time.
```

**Review by corvus-ch — CHANGES_REQUESTED** [review]

**Review by corvus-ch — APPROVED** [review]

**Pull request merged**